### PR TITLE
Story1 Review for Mike.

### PR DESCRIPTION
**Pull request by turnerror**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -24,5 +25,5 @@ function getDataFromDB(PDO $db)
 {
     $query = $db->query("SELECT * FROM `computerMice` WHERE (`name` IS NOT NULL AND `brand` IS NOT NULL);");
```
Comment (mporam) [inline functions.php]: `*` is bad practise, you should list out the field names. using `*` is really slow

File: functions.php [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
         if (!empty($mouse['weight'])) {
-            echo '<p>Weight: ' . $mouse["weight"] . 'g<p>';
+            $echo .= '<p>Weight: ' . $mouse["weight"] . 'g<p>';
         }
-        echo '</div>';
+        $echo .= '</div>';
     }
+    return $echo;
 }
 function linkDB()
```
Comment (mporam) [inline functions.php]: no return type

File: functions.php [hunk 3]
```diff
@@ -24,5 +25,5 @@ function getDataFromDB(PDO $db)
 {
```
Comment (mporam) [inline functions.php:27]: no return type on this function

File: test/displayMouseOnPageTest.php [hunk 4]
```diff
@@ -1,12 +1,10 @@
 <?php
-
 require_once('./functions.php');
 
 use PHPUnit\Framework\TestCase;
 
 class DisplayMouseOnPageTest extends TestCase
```
Comment (mporam) [inline test/displayMouseOnPageTest.php:6]: you havent got any success tests? how do you know this function works?

File: functions.php [hunk 5]
```diff
@@ -1,18 +1,19 @@
...
+            '<p>Brand: ' . $mouse["brand"] . '<p>';
         if (!empty($mouse['weight'])) {
-            echo '<p>Weight: ' . $mouse["weight"] . 'g<p>';
+            $echo .= '<p>Weight: ' . $mouse["weight"] . 'g<p>';
         }
-        echo '</div>';
+        $echo .= '</div>';
     }
+    return $echo;
 }
```
Comment (mporam) [inline functions.php:17]: missing line break between functions

**Review by mporam — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 6]
```diff
@@ -1,2 +1,3 @@
 .DS_Store
 .idea/
+index2.php
```
Comment (mporam) [inline .gitignore]: why?

File: test/displayMouseOnPageTest.php [hunk 7]
```diff
@@ -18,4 +18,13 @@ public function testThrowsErrorWhenGivenNoArgument()
 
         displayMouseOnPage();
     }
+    public function testSuccessDisplayMouseOnPage(){
```
Comment (mporam) [inline test/displayMouseOnPageTest.php]: opening `{` should be on its own line

**Review by mporam — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 6 above)
Comment (turnerror) [reply .gitignore]: sorry was testing something and it came up i should just delete the file haha

File: collectionproject_2020-03-23.sql [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;
+
+
+# Dump of table computerMice
+# ------------------------------------------------------------
+
+DROP TABLE IF EXISTS `computerMice`;
+
+CREATE TABLE `computerMice` (
+  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
```
Comment (mporam) [inline collectionproject_2020-03-23.sql]: looks like this is an old export of your database, there is no image field here?

**Review by mporam — CHANGES_REQUESTED** [review]

**Review by mporam — APPROVED** [review]

**Pull request merged**